packages/squiggle-lang/src/ast/peggyParser.peggy

```diff
@@ -269,10 +269,11 @@ unitIdentifier 'identifier'
 dollarIdentifier 'identifier'
   = ([\$_a-z]+[\$_a-z0-9]i*) { return h.nodeIdentifier(text(), location()); }
 
+escapeSeq = ("\\\\") { return "\\"; } / ('\\"') { return '"'; }  / ("\\'") { return "'"; }
 
 string 'string'
-  = characters:("'" @([^'])* "'") { return h.nodeString(characters.join(''), location()); }
-  / characters:('"' @([^"])* '"') { return h.nodeString(characters.join(''), location()); }
+  = "'" characters:(!("'" / "\\") . { return text(); } / escapeSeq)* "'" { return h.nodeString(characters.join(''), location()); }
+  / '"' characters:(!('"' / "\\") . { return text(); } / escapeSeq)* '"' { return h.nodeString(characters.join(''), location()); }
```

Comment:
1) Please add tests.
2) I'd suggest using `$` to pluck the text, instead of `return text();` (I'm not sure this matters, but I'm _slightly_ worried about parser performance; we should add benchmarks someday).
3) This could be simplified if we're willing to do unescaping in JS; see `delimitedComment` definition, the same pattern would be enough here. Something like `string = '"' characters:$(([\\]. / [^"])*) '"'`.
4) Unescaping in JS might be a good thing, I think, either with `.replaceAll`, or even with `JSON.parse` (are there any other standard ways to do that? I couldn't find any...).

Reply:
1. Sorry for leaving this as a non-draft, it very much is
2. I've read JS implementation in Peggy, and was amused that they don't use plucking as extensively as we do — so I read a bit more. In generated code it really does look like calling `text()` will incur additional costs, but I really want to check it out later too. 
3. I would probably not do `\\.` as that's not compatible with multi-character escape sequences. Alas I can understand frustration of seeing `\\\\ / \\\" / \\\'` and wanting to immediately change it — sorry again for not marking this as a draft)
4. Using JSON parse is a great idea! This brings me back to question in PR header: Should I add hex sequences, newlines, and the rest of JSON/JS?

Reply:
3) Wait, are you sure? Can you give an example that won't be parsed correctly? To be clear, I'm saying "do (3) and then do JSON.parse on top of it"; (3) by itself won't even unescape `\\`.
4) Yep, feature parity with JSON is important, we want Squiggle to be a superset of JSON; then people would be able to insert JSON strings to Squiggle. Parity with JS is less important; I'm not sure what's the difference between JSON and JS for double-quoted strings, tbh.

Reply:
3. I mean that `\u0000` will be resolved as a escape token `\u` and `0000`. Which is bad for highlighting at least (so I definitely won't be doing that in Lezer). It will still be functional, of course.

Reply:
Oh right, I didn't pay any attention to Lezer, only Peggy.

Reply:
Well, and the other problem I've encountered — it is really cumbersome to extract error locations from JSON.parse errors — only through parsing the message, which I am not sure is standardized.

Reply:
Yea - error locations are insanely tricky. We do a mediocre job with this in other places, I'd have a low bar for these for this PR.